src/modules/preview-site/components/preview-site-row.tsx

```diff
@@ -0,0 +1,86 @@
+import { Spinner } from '@wordpress/components';
+import { __, sprintf } from '@wordpress/i18n';
+import { useI18n } from '@wordpress/react-i18n';
+import { useEffect } from 'react';
+import { ArrowIcon } from 'src/components/arrow-icon';
+import Button from 'src/components/button';
+import { useExpirationDate } from 'src/hooks/use-expiration-date';
+import { useFormatLocalizedTimestamps } from 'src/hooks/use-format-localized-timestamps';
+import { useSnapshots } from 'src/hooks/use-snapshots';
+import { useUpdateDemoSite } from 'src/hooks/use-update-demo-site';
+import { getIpcApi } from 'src/lib/get-ipc-api';
+import { PreviewActionButtonsMenu } from './preview-action-buttons-menu';
+import { ProgressRow } from './progress-row';
+
+interface PreviewSiteRowProps {
+	snapshot: Snapshot;
+	selectedSite: SiteDetails;
+}
+
+export function PreviewSiteRow( { snapshot, selectedSite }: PreviewSiteRowProps ) {
+	const { __ } = useI18n();
+	const { url, date, isDeleting } = snapshot;
+	const { countDown } = useExpirationDate( date );
+	const { fetchSnapshotUsage } = useSnapshots();
+	const { isDemoSiteUpdating } = useUpdateDemoSite();
+	const isSiteDemoUpdating = isDemoSiteUpdating( snapshot.atomicSiteId );
```

Comment:
We could consider using the "preview" site here already:

```suggestion
	const isPreviewSiteUpdating = isDemoSiteUpdating( snapshot.atomicSiteId );
```

Reply:
Updated in https://github.com/Automattic/studio/pull/846/commits/e48a71ba37b80dddde02617ded05fc38636e324a